src/main/java/securbank/controller/ExternalUserController.java

```diff
@@ -80,56 +96,140 @@ public String newTransactionForm(Model model){
 		return "external/createtransaction";
 	}
 	
+	@GetMapping("/user/transaction/otp")
+	public String createTransactionOtp(Model model){
+		model.addAttribute("transaction", new Transaction());
+		logger.info("GET request: Extrernal user transaction generate OTP");
+		User currentUser = userService.getCurrentUser();
+		otpService.createOtpForUser(currentUser);
+		
+		return "redirect:/user/createtransaction";
+	}
+	
 	@PostMapping("/user/createtransaction")
     public String submitNewTransaction(@ModelAttribute Transaction transaction, BindingResult bindingResult) {
 		logger.info("POST request: Submit transaction");
 		
 		transactionFormValidator.validate(transaction, bindingResult);
 		
+		String otp = otpService.getOtpByUser(transaction.getAccount().getUser()).getCode();
```

Comment:
We are getting transaction from user input. So its account object will always be null. This is leading to a null pointer exception. Im guessing this should be `userService.getCurrentUser()` instead of `transaction.getAccount().getUser()`